lib/bugsnag.js

```diff
@@ -68,6 +68,17 @@ Bugsnag.register = function(apiKey, options) {
 Bugsnag.configure = function(options) {
     Configuration.configure(options);
 
+    // If notify release stages is set and we're not going to be notifying bugsnag we need to log
+    // this once and then prevent it from being logged in the future
+    if (!Bugsnag.shouldNotify()) {
+        Configuration.logger.warn(Configuration.apiKey ?
+            "Current release stage not permitted to send events to Bugsnag." :
+            "Bugsnag has not been configured with an api key!");
+        Configuration.logger.warn(Configuration.logAllDeliveryErrors ?
+            "Delivery errors will still be reported" :
+            "Delivery errors will not be reported. Configure logAllDeliveryErrors to change this behaviour");
```

Comment:
I think these log messages are confusing and could be worded better…

Something along the lines of:
"Calls to bugsnag.notify() will silently fail"
or
"Nothing will be sent to Bugsnag. Any bugsnag.notify() calls are essentially NOOPs."